abi/codecForCompositeTypes_option.go

```diff
@@ -0,0 +1,65 @@
+package abi
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+func (c *codec) encodeNestedOption(writer io.Writer, value OptionValue) error {
+	if value.Value == nil {
+		_, err := writer.Write([]byte{0})
+		return err
+	}
+
+	_, err := writer.Write([]byte{1})
+	if err != nil {
+		return err
+	}
+
+	return c.doEncodeNested(writer, value.Value)
+}
+
+func (c *codec) encodeTopLevelOption(writer io.Writer, value OptionValue) error {
+	if value.Value == nil {
+		return nil
+	}
+
+	_, err := writer.Write([]byte{1})
+	if err != nil {
+		return err
+	}
+
+	return c.doEncodeNested(writer, value.Value)
+}
+
+func (c *codec) decodeNestedOption(reader io.Reader, value *OptionValue) error {
+	bytes, err := readBytesExactly(reader, 1)
+	if err != nil {
+		return err
+	}
+
+	if bytes[0] == 0 {
+		value.Value = nil
+		return nil
+	}
+
+	return c.doDecodeNested(reader, value.Value)
+}
+
+func (c *codec) decodeTopLevelOption(data []byte, value *OptionValue) error {
+	if len(data) == 0 {
+		value.Value = nil
```

Comment:
This can happen if you have only one Option arg for endpoint, right?

Reply:
`if len(data) == 0` is to handle the `""`. Here, `data` is a "part of the transaction data" - a call to `decodeTopLevel*` handles one data part at a time. 

Is this all right?